pkg/controller/bastion/options.go

```diff
@@ -79,19 +83,31 @@ func DetermineOptions(bastion *extensionsv1alpha1.Bastion, cluster *controller.C
 		"Name": &baseResourceName,
 		"Type": to.StringPtr("gardenctl"),
 	}
-
+	infrastructureConfig := &api.InfrastructureConfig{}
+	if err = json.Unmarshal(cluster.Shoot.Spec.Provider.InfrastructureConfig.Raw, infrastructureConfig); err != nil {
+		return nil, err
+	}
+	// own vnet enabled if own vnet resourceGroup exist
+	myVnetEnabled := false
+	vNetrg := ""
+	if infrastructureConfig.Networks.VNet.ResourceGroup != nil && *infrastructureConfig.Networks.VNet.ResourceGroup != "" {
```

Comment:
I think a more appropriate way would to resolve the RG of the VNet is from the status e.g. [see here](https://github.com/gardener/gardener-extension-provider-azure/blob/d31fce706f474b8b1fe211dc22a882f3ba23a762/pkg/controller/worker/machines.go#L198)

Reply:
done remove 2 fields and pass it in `ensureNic` bdd779563772700d8883ace9ceb911f121f52817